### ISSUE TITLE
strategy/policy to support higher resolution for timestamp

### ISSUE THREAD

**Issue by koron** [go-xorm/xorm#1157]
Similar with #259

I want to use milliseconds (or higher) resolution for timestamp.
Current go-xorm/xorm supports just seconds resolution for `core.TimeStamp`.
https://github.com/go-xorm/xorm/blob/b07c4067034594d1a43e314b10153ff9cfc58f86/engine.go#L1570-L1571

I know that it is supported with timezone (`core.TimeStampz`).
https://github.com/go-xorm/xorm/blob/b07c4067034594d1a43e314b10153ff9cfc58f86/engine.go#L1572-L1577

But `core.TimeStampz` can't be used for my case.
So I'm going to implement it and make PR for go-xorm/xorm that choose resolution of `core.TimeStamp`.

My questions are:

* Do you interest (to merge) if I implemented option to use higher resolution for `core.TimeStamp`?
* Which is a preferred strategy/policy for go-xorm/xorm to implement this like features?
    1. Add an optional field to `Engine` (I'll start from this)
    2. Choose by dialect. 
    3. Or any others?

**Comment by koron** [comment]
No, it doesn't work.
I tried it already, and doesn't work for `created` or `updated` fields.

**Comment by lunny** [comment]
@koron right, it doesn't work with created and updated, but you can do it yourself via `BeforeInsert`, `AfterInsert`, `BeforeUpdate` and `AfterUpdate`.

**Comment by koron** [comment]
@lunny Do you mean that create `type CreatedAtTime Time` like type and define those methods, and use it instead of `createdat`?
